nameable.rb

```diff
@@ -0,0 +1,18 @@
+class Nameable
+  def correct_name
+    raise 'NotImplementedError'
+  end
+end
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```

Comment:
- [ ] It is not best practice to put two classes in the same file so kindly move the ```Decorator``` class to its own file.

Reply:
fixed